src/Commands/AliasesCommand.php

```diff
@@ -2,38 +2,266 @@
+     * @option string $target Base name to use to generate path to alias files.
+     * @option boolean $db-url Obsolete option included to preserve backwards compatibility. No longer needed.
      *
      * @return string|null
      *
      * @usage Saves Pantheon Drush aliases for sites on which the currently logged-in user is on the team to ~/.drush/pantheon.aliases.drushrc.php.
-     * @usage --print Displays Pantheon Drush aliases for sites on which the currently logged-in user is on the team.
-     * @usage --location=<full_path> Saves Pantheon Drush aliases for sites on which the currently logged-in user is on the team to <full_path>.
+     * @usage --print Displays Pantheon Drush 8 aliases for sites on which the currently logged-in user is on the team.
+     * @usage --location=<full_path> Saves Pantheon Drush 8 aliases for sites on which the currently logged-in user is on the team to <full_path>.
+     */
+    public function aliases($options = [
+        'print' => false,
+        'location' => null,
+        'all' => false,
+        'only' => '',
+        'type' => 'all',
+        'base' => '~/.drush',
+        'db-url' => true,
+        'target' => 'pantheon',
+    ])
+    {
+        // Be forgiving about the spelling of 'yaml'
+        if ($options['type'] == 'yaml') {
+            $options['type'] = 'yml';
+        }
+
+        $this->log()->notice("Fetching site information to build Drush aliases...");
+        $site_ids = $this->getSites($options);
+
+        // Collect information on the requested sites
+        $collection = $this->getAliasCollection($site_ids);
+
+        $this->log()->notice("{count} sites found.", ['count' => count($site_ids)]);
+
+        // Write the alias files (only of the type requested)
+        $emitters = $this->getAliasEmitters($options);
+        if (empty($emitters)) {
+            throw new \Exception('No emitters; nothing to do.');
+        }
+        foreach ($emitters as $emitter) {
+            $this->log()->debug("Emitting aliases via {emitter}", ['emitter' => get_class($emitter)]);
+            $this->log()->notice($this->shortenHomePath($emitter->notificationMessage()));
+            $emitter->write($collection);
+        }
+    }
+
+    /**
+     * Utility function to convert references to the home path to simply '~'
+     *
+     * @param string $message
+     *
+     * @return string
+     */
+    protected function shortenHomePath($message)
+    {
+        return str_replace($this->getConfig()->get('user_home'), '~', $message);
+    }
+
+    /**
+     * Fetch those sites indicated by the commandline options.
+     *
+     * @param array $options
+     *   Full set of commanline options, some of which may affect selected set
+     *   of sites returned.
+     *
+     * @return array
+     *   Associative array of site id => site name
      */
-    public function aliases($options = ['print' => false, 'location' => null,])
+    protected function getSites($options)
```

Comment:
The functions should be organized by visibility (public, then protected, then private), then alphabetized within the group.